### PR TITLE
Release build

### PR DESCRIPTION
**Pull request by rahulbhivare**
Fixed memory corruption issue when sdk is build in release mode

### CODE REVIEW

File: lib/cpp/antaris_sdk_environment.cc [hunk 1]
```diff
@@ -107,10 +107,19 @@ static void parse_a_conf(char *conf_line, conf_t *out_conf)
 
 static void update_a_conf(char *conf_line)
 {
+    int8_t length = -1;
     conf_t a_conf = {0};
 
     parse_a_conf(conf_line, &a_conf);
 
+    do {
```
Comment (deepaktawri) [inline lib/cpp/antaris_sdk_environment.cc:115]: can we use len = strlen(a_conf.value) to do the same thing ? If len is greater than MAX len, its error.

Comment (deepaktawri) [reply lib/cpp/antaris_sdk_environment.cc:115]: I had revised my comment to use strnlen_s. Can you please check the comment again.

File: lib/cpp/antaris_sdk_environment.cc [hunk 2]
```diff
@@ -107,9 +106,16 @@ static void parse_a_conf(char *conf_line, conf_t *out_conf)
...
 static void update_a_conf(char *conf_line)
 {
+    int8_t length = -1;
     conf_t a_conf = {0};
+    size_t length = 0;
 
     parse_a_conf(conf_line, &a_conf);
+    length = strnlen(a_conf.value, MAX_IP_OR_PORT_LENGTH);
+    if(length == MAX_IP_OR_PORT_LENGTH) {
+        printf("Property value is greater than %d characters is not supported. Going with default value \n", MAX_IP_OR_PORT_LENGTH);
```
Comment (deepaktawri) [inline lib/cpp/antaris_sdk_environment.cc:116]: Can we ASSERT here in #ifdef DEBUG mode because we never expect such error in config file.

**Review by deepaktawri — APPROVED** [review]

**Comment by rolandjitsu** [comment]
Looking fwd to see this merged, thanks.

File: lib/cpp/antaris_sdk_environment.cc [hunk 3]
```diff
@@ -107,9 +106,16 @@ static void parse_a_conf(char *conf_line, conf_t *out_conf)
 
 static void update_a_conf(char *conf_line)
 {
+    int8_t length = -1;
     conf_t a_conf = {0};
+    size_t length = 0;
```
Comment (rolandjitsu) [inline lib/cpp/antaris_sdk_environment.cc:111]: This is causing cmake to error when building our app:
```
120.1 [ 27%] Built target paapis_lib
120.4 [ 30%] Building CXX object _deps/antaris-sdk-build/CMakeFiles/antaris_api.dir/lib/cpp/antaris_sdk_environment.cc.o
120.9 [ 32%] Linking CXX static library ../../../lib/libsignals.a
121.3 /code/build/_deps/antaris-sdk-src/lib/cpp/antaris_sdk_environment.cc: In function 'void update_a_conf(char*)':
121.3 /code/build/_deps/antaris-sdk-src/lib/cpp/antaris_sdk_environment.cc:111:12: error: conflicting declaration 'size_t length'
121.3   111 |     size_t length = 0;
121.3       |            ^~~~~~
121.3 /code/build/_deps/antaris-sdk-src/lib/cpp/antaris_sdk_environment.cc:109:12: note: previous declaration as 'int8_t length'
121.3   109 |     int8_t length = -1;
121.3       |            ^~~~~~
121.5 make[2]: *** [_deps/antaris-sdk-build/CMakeFiles/antaris_api.dir/build.make:118: _deps/antaris-sdk-build/CMakeFiles/antaris_api.dir/lib/cpp/antaris_sdk_environment.cc.o] Error 1
```

https://github.com/antaris-inc/SatOS-Payload-SDK/pull/25 should fix it (testing that right now).

**Pull request merged**